ably/realtime_channel.go

```diff
@@ -585,25 +585,42 @@ func (em ChannelEventEmitter) OffAll() {
 }
 
 // Publish publishes a single message to the channel with the given event name and message payload.
-// A callback may optionally be passed in to this call to be notified of success or failure of the operation.
-// When publish is called with this client library, it won't attempt to implicitly attach to the channel,
-// so long as transient publishing is available in the library. Otherwise, the client will implicitly attach (RTL6i).
-//
-// This implicitly attaches the channel if it's not already attached. If the
-// context is canceled before the attach operation finishes, the call
-// returns with an error, but the operation carries on in the background and
-// the channel may eventually be attached and the message published anyway.
 func (c *RealtimeChannel) Publish(ctx context.Context, name string, data interface{}) error {
```

Comment:
Publish publishes a single message to the channel with the given event name and message payload
Publish call remains blocked till the message is published, it returns an error with nil/non-nil value representing success/failure respectively.

Reply:
Can you update the same doc for `PublishMultiple` ?

Reply:
How about
```
// Publish publishes a single message to the channel with the given event name and message payload.
//
// This will block until either the publish is acknowledged or fails to deliver.
```

> it returns an error with nil/non-nil value representing success/failure respectively.

That seems t bit redundant?

Reply:
// Publish publishes a single message to the channel with the given event name and message payload.
// This will block until either the publish is acknowledged (returns error as nil) or fails to deliver ( returns error value )
//If the context is cancelled before the attach operation finishes, the call returns an error but the publish will carry on in the background and may eventually be published anyway.
WDYT?

Reply:
sure updated